### ISSUE TITLE
[Cellpose-SAM] Overlapping instance masks are lost when generating the final masks output

### ISSUE THREAD

**Issue by huangjin520** [MouseLand/cellpose#1190]
### **❓ Description**
Hi team,

Thank you for the great work on integrating SAM into Cellpose! I’ve been testing Cellpose-SAM for DAPI-stained nuclei segmentation and noticed a potential limitation with overlapping instances.

### **✅ Reproduction steps**
Run `model.eval(img)` on a DAPI image with dense nuclei.
You will observe that in crowded regions, instances are missing or merged, and overlap between instances is lost.

### **🔍 My Questions:**
Is this behavior expected? Is there a way to access all original binary instance masks before the final flattening?

### **💡How to save overlapping regions' information  (if applicable)**
If possible, could you add an option in eval() to return:

```python
masks, flows, styles, masks_list  # ← each element in masks_list is a binary mask
```

Or alternatively, provide a utility function to extract the raw masks from SAM before merging.

Thank you for your time and the amazing tool!

Best regards,

![Image](https://github.com/user-attachments/assets/d6bc0d99-ed16-43cb-8228-7aab077d61c2)
**

![Image](https://github.com/user-attachments/assets/09690d25-26bf-4e18-b4e8-745b522a4933)

**
`masks.shape =[256.256]`

**Comment by mrariden** [comment]
Due to the flows representation that the network predicts, Cellpose is not capable of predicting overlapping masks. You can read more about this in the papers, particularly the first cellpose paper.
